contrib/mod_ldap.c

```diff
@@ -1257,23 +1257,28 @@ static struct group *pr_ldap_group_lookup(pool *p, char *filter_template,
   return gr;
 }
 
+static void cached_quota_cleanup(void *event_data) {
+  cached_quota = NULL;
+}
+
 static void parse_quota(pool *p, const char *replace, char *str) {
   char **elts, *token;
 
   if (cached_quota == NULL) {
     cached_quota = make_array(p, 9, sizeof(char *));
+    register_cleanup2(p, NULL, cached_quota_cleanup);
```

Comment:
At some point, I start to wonder whether having this `cached_quota` pointer makes sense at all.  The code flow would certainly be simpler if it was removed entirely.  And the mod_quotatab module does not look up the quota limits for users that often, I feel.